yarn-project/p2p/src/util.ts

```diff
@@ -141,3 +144,46 @@ export async function configureP2PClientAddresses(
 
   return config;
 }
+
+/**
+ * Get the peer id private key
+ *
+ * 1. Check we have a peer id private key persisted in the node
+ * 2. If not, check if we have a peer id private key in the config
+ * 3. If not, create a new one, then persist it in the node
+ *
+ */
+export async function getPeerIdPrivateKey(config: { peerIdPrivateKey?: string }, store: AztecKVStore): Promise<string> {
+  const peerIdPrivateKeySingleton: AztecSingleton<string> = store.openSingleton('peerIdPrivateKey');
```

Comment:
Should we prioritise a configured value over a stored one? Currently, once a key is stored, it's impossible to change that key without wiping the data directory.